### ISSUE TITLE
Proof title replaces "Proof" in PDF output

### ISSUE THREAD

**Issue by cwickham** [quarto-dev/quarto-cli#8224]
### Bug description

We [document](https://quarto.org/docs/authoring/cross-references.html#theorems-and-proofs) for proofs that:
> As with theorems you can optionally include a heading as the first element of the div (or a name attribute) to give the environment a caption for typesetting (this typically appears in parentheses after the environment title).

But that doesn't seem to be true for PDF output. I don't think this is a regression, since it also doesn't work in 1.3.450. 

### Steps to reproduce

````qmd
---
title: "Proofs"
format: pdf
---

::: {.proof}

## Title

Proof with title

:::

::: {.proof}

Proof without title

:::

````

### Expected behavior

Something like `format: html`, the output begins with i.e. Proof (Title)
<img width="202" alt="Screenshot 2024-01-11 at 2 41 14 PM" src="https://github.com/quarto-dev/quarto-cli/assets/25964/755de537-3216-487a-97bf-f97665391131">

### Actual behavior

The title appears, but "Proof" doesn't
<img width="124" alt="Screenshot 2024-01-11 at 2 40 54 PM" src="https://github.com/quarto-dev/quarto-cli/assets/25964/6f62a3ff-1004-4ff7-a462-34377b4ab729">

### Your environment

- OS: Mac 14.0 Sonoma

### Quarto check output

```bash
Quarto 99.9.9
[✓] Checking versions of quarto binary dependencies...
      Pandoc version 3.1.11: OK
      Dart Sass version 1.69.5: OK
      Deno version 1.37.2: OK
[✓] Checking versions of quarto dependencies......OK
[✓] Checking Quarto installation......OK
      Version: 99.9.9
      Path: /Users/wickhamc/Documents/posit/quarto-cli/package/dist/bin

[✓] Checking tools....................OK
      TinyTeX: v2023.09
      Chromium: (not installed)

[✓] Checking LaTeX....................OK
      Using: TinyTex
      Path: /Users/wickhamc/Library/TinyTeX/bin/universal-darwin
      Version: 2023

[✓] Checking basic markdown render....OK

[✓] Checking Python 3 installation....OK
      Version: 3.12.0
      Path: /usr/local/bin/python3
      Jupyter: 5.5.0
      Kernels: ir, julia-1.9, python3

[✓] Checking Jupyter engine render....OK

[✓] Checking R installation...........OK
      Version: 4.3.2
      Path: /Library/Frameworks/R.framework/Resources
      LibPaths:
        - /Library/Frameworks/R.framework/Versions/4.3-arm64/Resources/library
      knitr: 1.45
      rmarkdown: 2.21

[✓] Checking Knitr engine render......OK
```

**Comment by cderv** [comment]
It seems the HTML layout for the `Proof` environment available through LaTeX amsthm may not be correct.

LaTex for proof is

````latex
\begin{proof}[Custom Title]
To prove it by contradiction try and assume that the statement is false,
proceed from there and at some point you will arrive to a contradiction.
\end{proof}
````

gives 
![image](https://github.com/quarto-dev/quarto-cli/assets/6791940/e5a6d087-9b1e-4b10-a31a-77e64e6667ea)


which seems expected according to amsthm documentation. And this is what we create from the div syntax for PDF output

Not sure why we do output differently for HTML. We output exactly the same for HTML in **bookdown** (https://bookdown.org/yihui/bookdown/markdown-extensions-by-bookdown.html#theorems). So maybe there is a good reason, or this was just a port of what has been done in R Markdown ecosystem. 

It seems like we should maybe change HTML or at least update the documentation

**Comment by cscheid** [comment]
Do you want me to change this for 1.4? The issue is that `proof` is specifically different from other solution environments (see also the QED symbol). That has been the case in 1.3.

**Comment by cscheid** [comment]
(cc @gordonwoodhull, since you're working on Typst theorems) I don't think it's feasible for us in the long term to emit content that's exactly equal across the styles.

I mean that more generally than just this specific case. I mean that we already don't emit consistent output for things like callouts, because we prefer the "native" appearance on formats that have a sufficiently-similar notion.

In the long run, forcing identical output for all formats means a number of things we don't want:

- we will be bound by the "lowest common denominator" across all formats, and a new format can only at best be as good as our worst format for any given feature
- we take over the responsibility of rendering this feature unconditionally into the future
  - that means we take in a lot of additional work, 
  - and that we can't track improvements on formats we target

Maybe as we implement a feature in a given format, we have to make an explicit internal call if that is going to be a "native" feature or a "borrowed" feature. For example, Callouts are native in HTML but borrowed in GFM and docusaurus. Proofs are native in HTML, but borrowed in PDF and Typst, etc.

**Comment by cwickham** [comment]
Right, I think that is a reasonable approach. We should make it clear to users when and where these differences might occur. Perhaps, our format/feature matrix will help a lot with that.

**Comment by gordonwoodhull** [comment]
Nod, there are [many stylistic differences](https://github.com/quarto-dev/quarto-cli/pull/8825#issuecomment-1959734415) between renderings of theorems in LaTeX and Typst.

Is the idea that all `dev-docs/feature-format-matrix/qmd-files` will be rendered in each supported format and deployed to quarto.org? It would be nice to offer a preview e.g. in an iframe below the matrix, when you click on (another icon in?) a cell. (If we get that working we could also consider side-by-side comparison.)

**Comment by cscheid** [comment]
I don't think we've decided yet, but something of that sort would be nice. The issue with "just" an embedded iframe is that it won't work for .docx, .pptx, etc.

**Comment by gordonwoodhull** [comment]
We could offer download links for unembeddable formats, but yeah this is probably not the only difficulty with the iframe idea.

Perhaps for a first pass we could get them built and deployed and have links in the matrix, probably not trivial in itself.

**Comment by ute** [comment]
A quick fix for making proofs in pdf consistent with html would be to patch amsthm's proof environment:
```latex
\makeatletter
\renewenvironment{proof}[1][]{\par\pushQED{\qed}%
\normalfont\topsep6\p@\@plus6\p@\relax\trivlist%
\item[\hskip\labelsep\itshape{%
\proofname\ifx&#1&\relax\else~(#1)\fi}\@addpunct{.}]\ignorespaces}  
{\popQED\endtrivlist\@endpefalse}
\makeatother
```

**Comment by cscheid** [comment]
@ute The problem with that is that amsthm is very popular and produces "expected output". We definitely wouldn't want to do that by default, since it would break a number of PDF journal templates

**Comment by ute** [comment]
@cscheid This makes sense! - then there should be a quick fix the other way round, for those who want consistency between web and pdf appearance. I made the quick fix because I am writing lecture notes, and it would look weird either as pdf or as html. I consider to upload it as filter.
